config/src/config_wrapper.cpp

```diff
@@ -118,98 +236,218 @@ int32_t get_station_dist(int32_t from, int32_t to)
+
+    endstations.insert(endstations.end(), stations.begin(), stations.end());
+    endstations.insert(endstations.end(), vias.begin(), vias.end());
+
+    std::vector<bool> visited;
+
+    for (const auto &waypoint : endstations) {
+        visited.push_back(std::find(visited_waypoints.begin(), visited_waypoints.end(), waypoint) !=
+                          visited_waypoints.end());
     }
+
+    return visited;
 }
 
 void waypoint_visited(int32_t number_of_waypoints, int8_t *arr)
 {
     load();
     try {
-        static auto tmp = dynamic_config.get<std::vector<bool>>("waypoint_visited");
+        static auto tmp = convert_visited_waypoints();
         for (int i = 0; i < number_of_waypoints; i++) {
             arr[i] = tmp.at(i);
         }
     }
     catch (const std::exception &e) {
+        _log << "waypoint_visited";
         _log << e.what();
     }
 }
 
-void station_waypoint(int32_t number_of_stations, int32_t *arr)
+static std::vector<int> get_station_list()
+{
+    auto stations = static_config.get<std::vector<int>>("stations");
+    auto endstations = static_config.get<std::vector<int>>("end_stations");
+
+    endstations.insert(endstations.end(), stations.begin(), stations.end());
+
+    return endstations;
+}
+
+void station_list(int32_t number_of_stations, int32_t *arr)
 {
     load();
     try {
-        static auto tmp = static_config.get<std::vector<int>>("station_waypoint");
+        static auto tmp = get_station_list();
         for (int i = 0; i < number_of_stations; i++) {
             arr[i] = tmp.at(i);
         }
     }
     catch (const std::exception &e) {
+        _log << "station_list";
         _log << e.what();
     }
 }
 
-void waypoint_schedule(int32_t number_of_stations, int32_t number_of_robots, int32_t *arr)
+int32_t convert_to_action(std::pair<std::string, int> pair)
+{
+    if (pair.first.compare("Waypoint") == 0) {
+        return WAYPOINT;
+    }
+    else if (pair.first.compare("Hold") == 0) {
+        return HOLD;
+    }
+
+    return DONE;
+}
+
+static std::vector<std::vector<std::pair<std::string, int>>> get_waypoint_plan()
+{
+    static auto tmp = dynamic_config.get<std::vector<std::vector<std::pair<std::string, int>>>>(
+        "robot_info_map", "waypoint_plan");
+    return tmp;
+}
+
+int32_t get_next_action_type(int32_t robot, int32_t step)
 {
     load();
     try {
-        /*auto tmp = dynamic_config.get<std::vector<std::vector<int>>>("station_schedule");
-        for (int i = 0; i < num_robots - 1; i++) {
-          for (int j = 0; j < number_of_stations; j++) {
-            arr[i * number_of_stations + j] = tmp.at(i).at(j);
-          }
-        }*/
+        static auto tmp = get_waypoint_plan();
+        // Other robot index start at 2.
+        // Therefore, we subtract 2.
+        auto robot_schedule = tmp.at(robot - 2);
+
+        return static_cast<int>(robot_schedule.size()) > step ? convert_to_action(robot_schedule.at(step)) : DONE;
```

Comment:
Hvorfor/hvornår er der behov for at cast?

Reply:
Fordi vi sammenligner `size_t` og `int32_t`, så clang giver en warning, og fordi vi bruger -Werror, så bliver det til en fejl.